src/GovUk.Education.ExploreEducationStatistics.Publisher/Functions/StageScheduledReleasesFunction.cs

```diff
@@ -36,38 +45,100 @@ public StageScheduledReleasesFunction(IQueueService queueService, IReleasePublis
             ExecutionContext executionContext,
             ILogger logger)
         {
-            logger.LogInformation("{0} triggered at: {1}",
+            logger.LogInformation("{FunctionName} triggered at: {DateTime}",
                 executionContext.FunctionName,
                 DateTime.UtcNow);
 
-            await PublishReleaseFilesAndStageContent();
+            await PublishReleaseFilesAndStageContent((await QueryScheduledReleasesForToday()).ToArray());
 
             logger.LogInformation(
-                "{0} completed. {1}",
+                "{FunctionName} completed.  Will be scheduled again to run at {NextDateTime}",
                 executionContext.FunctionName,
                 timer.FormatNextOccurrences(1));
         }
 
-        private async Task PublishReleaseFilesAndStageContent()
+        /// <summary>
+        /// Azure function which triggers publishing files and staging content for all Releases that are scheduled to
+        /// be published later during the day. This is triggered manually by an HTTP post request, and is disabled in
+        /// production environments.
+        /// </summary>
+        /// <param name="request">
+        /// An optional JSON request body with a "ReleaseIds" array can be included in the POST request to limit the
+        /// scope of the Function to only the provided Release Ids.
+        /// </param>
+        /// <param name="executionContext"></param>
+        /// <param name="logger"></param>
+        [FunctionName("StageScheduledReleasesImmediately")]
```

Comment:
I was considering that this endpoint could be removed, and the staging and publishing happens in a single function instead. I think you're probably right to keep them separate - that means we can test the stages are displayed correctly between the two requests - but part of me does like the simplicity of one endpoint. 

Not suggesting any changes here - just felt compelled to comment. Don't know if you have any additional thoughts about it? It's possible there would be a complication over waiting for the staging to complete?

Reply:
Yeah, it's a point I've discussed with Ben and Nick as well recently.  The staging step could actually potentially go away completely and we fall back on lazy caching to generate the content that is being staged here.  We'd still need something to copy the files over (PublishReleaseFilesFunction) which also happens at midnight, but hopefully we'll be able to make this disappear too soon.

Reply:
Sweeeeet